src/_zkapauthorizer/replicate.py

```diff
@@ -23,14 +23,77 @@
 ]
 
 from collections.abc import Awaitable
+from io import BytesIO
 from typing import BinaryIO, Callable
 
+import cbor2
+from attrs import frozen
 from twisted.python.lockfile import FilesystemLock
 
 from .config import REPLICA_RWCAP_BASENAME
 from .tahoe import Tahoe, attenuate_writecap
 
 
+@frozen
+class Change:
+    """
+    Represent an item in a replication event stream
+    """
+
+    sequence: int  # the sequence-number of this event
+    statement: str  # the SQL statement string
+
+
+@frozen
+class EventStream:
+    """
+    A series of database operations represented as `Change` instances.
+    """
+
+    changes: tuple[Change]
+
+    def highest_sequence(self):
+        """
+        :return int: the highest sequence number in this EventStream (or
```

Comment:
I guess `Optional[int]`.  Also I've been slowly moving type info from the docstrings into annotations, so maybe this could move similarly.